### ISSUE TITLE
Record playlist JS1 Week 1

### ISSUE THREAD

**Issue by Dedekind561** [CodeYourFuture/JavaScript-Core-1-Coursework-Week1#468]
Assignees can add a plan here!

**Comment by Dedekind561** [comment]
Feedback for [first recording](https://us02web.zoom.us/rec/play/HkgeJoQrK5JzOERVumFnjkR12ugAhjYNSalZGVcezIOK5Tx6nzGWBjWkdyJx0Ra0pmEj3iFOEgmbgQgj.nDsLCM07X_47tyHr?startTime=1674838395000&_x_zm_rtaid=h0scTsAnS9OprffmaU4WAA.1675084447958.0130f9fedf52396ed4bdbfdd077cebd0&_x_zm_rhtaid=659) @gregdyke

**Comment by gregdyke** [comment]
2nd recording (much less successful than the 1st, but a better version would cover similar material):

- expressions vs statements 
- REPL
- String expressions
- Number expressions

If we're avoiding `+` for strings (sounds like a good thing), does that mean we go for `${greeting} Greg` ?

I agree with all your points. Except maybe github desktop as I'm not sure they are working in a checked out file at this point?

Function is intentionally deferred to a later video on functions (though it's a word that is used so it should go on the sticky note)

**Comment by gregdyke** [comment]
BTW, other videos in the series would be:
- defining and calling functions
- npm install + running tests
- non-directly-essential things (briefly covering the non-used version for concatenating strings, const vs let, possibly typeof as it's kind of useful to reflectively explore stuff, but otherwise not something needed)

**Comment by shieldo** [comment]
I like these a lot, especially the clarity given by using the REPL - that seemed very successful. I had the same issue with the explanation for what `node` refers to - trainees may remember back to the program that they would have installed in preparation for the module, so it would be nice to refer back to that when making an explanation.

It might be nice to mention you can use the up arrow in the terminal to bring up the previously-entered command to run again, rather than typing out the exact same command repeatedly. I expect there is a good argument for encouraging a sufficient fluidity in the terminal/ not needing to feel like it's a slog (especially as people will very often make typos when retyping!). Using a program like [Keycastr](https://github.com/keycastr/keycastr) might be useful here for displaying the keys you are entering for the screencast - I haven't experimented with this, though, and don't know whether you can filter the keypresses that show up.

I think I noticed a few possible errata on the micro level but some of these are likely slips of the tongue that wouldn't happen in a new version!

**Comment by Dedekind561** [comment]
Hello folks, 
Excellent work here!
In summary, I think videos 1, 2, 3 are ready to go in the playlist ✅ 
I think Video 4 could be edited now if you've got time and I'll explain why in the feedback below
I've left additional feedback for all the videos but to repeat any critiques for 1 to 3 are not blocking at this stage.

**Comment by Dedekind561** [comment]
#Video 1


## Positives
- Asking questions and prediction workflow works so well here. This is an excellent exemplar for volunteers who are going to record in the future. 


## Suggestions ( for future)
- When terminal is first opened, run `pwd` and `ls` commands to introduce/recap some basic CLI commands. Super useful for trainees to orient themselves in the terminal.
- Clarification that `=` means assignment operator and not equals as in maths

**Comment by Dedekind561** [comment]
### Video 2
Runtime: 17 mins 
[Link(https://us02web.zoom.us/rec/share/BEn9TTVl8DjoFE57ba3WDpWxxPYoR4tN9qZKakSbXpbFFNRBpslZql794JQt_w1-.QjVj-mpisrfipThI?startTime=1675618730000) 
Passcode: H3w#FRCF

#### Positives
- Good links back to prior knowledge
- Good extra detail on side effects/expressions/statements


#### Suggestions ( for future)
- Maybe use word "input" instead of "argument" before section on function calls and declaring functions.  
- Word "evaluate" used without definition
- Show REPL before the script as trainees can see expressions evaluated and printed without running `console.log` or executing a script
- `let greeting = 'hello'` executed in REPL without a `;` - something some trainees may notice
- At one point, you said all statements are expressions. Is this strictly true? A `for` loop is a statement but if it was also an expression then that implies the `for` statement evaluates to a single value, which could in turn be passed around.

**Comment by Dedekind561** [comment]
### Video 3
Runtime: 26 mins 
[Link](https://us02web.zoom.us/rec/play/kehi4inhi5rPlCU3RsXcgJLc9lsAFQMmANaPVCGR7m9PUZ4O0F9KHGPpce8ydV8NCGSuIIi5HxdrRrLj.7pcbh_3eLYNo3i6b?startTime=1675620113000&_x_zm_rtaid=AfuBeYt-S4qO7XuQH3bCUg.1675786291551.acf46eec3a7912209fe9b1bd6e218f90&_x_zm_rhtaid=591) 
Passcode: H3w#FRCF

#### Positives
- Excellent consultation of docs
- Excellent reference to `Math.round` - super intuitive function
- Great moment when you asked what would happen when there were no function calls in the script 
- Made good use of the `ReferenceError` for `add` when using the REPL


#### Suggestions ( for future)
- Show `Math.round` in the REPL first; then `Math.round()` and then finally `Math.round(8.7)`.
- Word "interpolation" used without definition
- Consider invoking `add1(100)` but _not_ putting a parameter in the function declaration. Ask what problem this presents us with?
- Emphasise that `Math.round` is in-built in contrast with `add1` or that `Math.round` can be referenced without any declaration or something along those lines

**Comment by Dedekind561** [comment]
### Video 4
Runtime: 26 mins 
[Link](https://us02web.zoom.us/rec/share/BEn9TTVl8DjoFE57ba3WDpWxxPYoR4tN9qZKakSbXpbFFNRBpslZql794JQt_w1-.QjVj-mpisrfipThI?startTime=1675622178000) 
Passcode: H3w#FRCF

#### Positives
- Excellent consultation of docs
- Excellent reference to `Math.round` - super intuitive function
- Great moment when you asked what would happen when there were no function calls in the script 
- Made good use of the `ReferenceError` for `add` when using the REPL


#### Suggestions ( for future)
- Show `Math.round` in the REPL first; then `Math.round()` and then finally `Math.round(8.7)`.
- Word "interpolation" used without definition
- Consider invoking `add1(100)` but _not_ putting a parameter in the function declaration at first. Ask what problem this presents us with?
- Emphasise that `Math.round` is in-built in contrast with `add1` or that `Math.round` can be referenced without any declaration or something along those lines

**Comment by gregdyke** [comment]
I assume we wouldn't do the restructuring in the discussion before running JS1 for this cohort?

**Comment by gregdyke** [comment]
This assumption is based on all our current exercises needing functions

**Comment by Dedekind561** [comment]
> I assume we wouldn't do the restructuring in the discussion before running JS1 for this cohort?

Yes absolutely, I don't think we should overcomplicate things at this stage.
